include/aws/http/request_response.h

```diff
@@ -414,6 +414,18 @@ int aws_http_headers_get(
     struct aws_byte_cursor name,
     struct aws_byte_cursor *out_value);
 
+/**
+ * Test if header name exists or not in headers
+ */
+AWS_HTTP_API
+bool aws_http_has_header_in_headers(const struct aws_http_headers *headers, struct aws_byte_cursor name);
+
+/**
+ * Test if header name exists or not in http message
+ */
+AWS_HTTP_API
+bool aws_http_has_header_in_http_message(struct aws_http_message *message, struct aws_byte_cursor name);
```

Comment:
Let's not bother adding this function to `aws_http_message`.
Someone can always aws_http_message_get_headers() and then operate on that.

The reason there's overlap between `aws_http_message` and `aws_http_headers` is that the "message" class existed first, and the "headers" class came later, but we kept the existing functions on the "message" class for API compatibility.